shared/modules/doc/README.md

```diff
@@ -1,53 +1,67 @@
 
-In order to maintain code agnostic from the inception of the implementation, protobuf3 is utilized for all structures in this package.
+> The bigger the interface, the weaker the abstraction.
 
-It is important to note, that while Pocket V1 strives to not share objects between modules, the genesis module will inevitably overlap between other modules.
+These interfaces that can be embedded in modules are defined in `shared/modules/module.go`. GoDoc comments will provide you with more information about each interface.
 
-Another architecture worth considering and perhaps is more optimal as the project nears mainnet is allowing each module to create and maintain their own genesis object and config files
+### **Module**
 
-### Code Organization
+A module is an abstraction (go interface) of a self-contained unit of functionality that carries out a specific task/set of tasks with the idea of being reusable, modular, testable and having a clear and concise API. A module might implement 0..N common interfaces. You can find additional details in the [Modules in detail](#modules-in-detail) section below.
 
-```bash
-genesis
-├── docs
-│   ├── CHANGELOG.md    # Genesis module changelog
-│   ├── README.md       # Genesis module README
-├── proto
-│   ├── account.proto   # account structure
-│   ├── actor.proto     # actor structure
-│   ├── config.proto    # configuration structure
-│   ├── gov.proto       # params structure
-│   ├── state.proto     # genesis state structure
-├── test_artifacts      # the central point of all testing code (WIP)
-│   ├── generator.go    # generate the genesis and config.json for tests and build
-│   ├── gov.go          # default testing parameters
-```
+### **Module mock**
+
+A mock is a stand-in, a fake or simplified implementation of a module that is used for testing purposes. It is used to simulate the behaviour of the module and to verify that the module is interacting with other modules correctly. Mocks are generated using `go:generate` directives together with the [`mockgen` tool](https://pkg.go.dev/github.com/golang/mock#readme-running-mockgen).
+
+### **Base module**
 
-TODO(#235): Update once runtime configs are implemented
+A base module is a module that implements a common interface, exposing the most basic logic. Base modules are meant to be **embedded** in module structs which implement this common interface **and** don't need to override the respective interface member(s). The intention being to improve DRYness (Don't Repeat Yourself) and to reduce boilerplate code. You can find the base modules in the `shared/modules/base_modules` package.
 
-### Module Typical Usage Example
+## Code Organization
+
+```bash
+├── base_modules            # All the base modules are defined here
+├── doc			            # Documentation for the modules
```

Comment:
NIT: Replace tabs with spaces.

Can't believe I'm saying this... https://www.youtube.com/watch?v=SsoOG6ZeyUI

Reply:
eye_twitches.gif

mad_men_meditate_explosion.gif

I am not gonna start a fight with you **now**  😉

Reply:
At least it would be over a good reason